src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+
+/**
+ * Parses in the user input and executes the corresponding actions
+ */
+public class Parser {
+    private final String command;
+    private final Ui ui;
+
+    /**
+     * Constructor for parser
+     * @param command user's inputted command
+     */
+    public Parser(String command) {
+        this.command = command;
+        this.ui = new Ui();
+    }
+
+    /**
+     * Parses the command and matches it with the corresponding action to be taken
+     * Executes the action on the task list given
+     * @param tasks ArrayList of current tasks stored on the chatbot
+     * @return boolean true if bye command entered and chatbot should be stopped, false otherwise
+     */
+    public String executeCommand(TaskList tasks, Storage storage) throws IOException {
+        String[] commands = command.strip().split(" ", 2);
+        try {
+            switch (commands[0]) {
+            case "bye":
+                storage.saveTasks(tasks);
+                return ui.showBye();
+            case "list":
+                return ui.getCorrectMessage(Ui.Commands.LIST, tasks, 0);
+            case "mark":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("mark");
+                }
+                int indexMark = Integer.parseInt(commands[1]) - 1;
+                tasks.markIndex(indexMark);
+                return ui.getCorrectMessage(Ui.Commands.MARK, tasks, indexMark);
+            case "unmark":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("unmark");
+                }
+                int indexUnmark = Integer.parseInt(commands[1]) - 1;
+                tasks.unmarkIndex(indexUnmark);
+                return ui.getCorrectMessage(Ui.Commands.UNMARK, tasks, indexUnmark);
+            case "todo":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("todo");
+                }
+                tasks.add(new Todo(commands[1]));
+                return ui.getCorrectMessage(Ui.Commands.TASK, tasks, tasks.getSize() - 1);
+            case "event":
+                if (commands.length == 1) {
```

Comment:
This entire switch case statement is a little bit unreadable, I would consider adding more vertical spacing and seeing where you can abstract out some of your logic. 

for example, does it make sense for you to do your error check and throwing here? or should they be abstracted into your classes? `Ask Don't Tell`.